### PR TITLE
Tidy up the error pages

### PR DESCRIPTION
**Pull request by paf31**
This tries to update the error pages to use a consistent style, and remove out-of-date content.

- [x] CannotWriteFile
- [x] CycleInDeclaration
- [x] CycleInModules
- [x] DuplicateModule
- [x] ErrorParsingFFIModule
- [x] ErrorParsingModule
- [ ] EscapedSkolem
- [ ] ExprDoesNotHaveType
- [ ] KindsDoNotUnify
- [ ] MissingFFIModule
- [ ] NoInstanceFound
- [ ] OrphanInstance
- [ ] OverlappingArgNames
- [ ] OverlappingPattern
- [x] ScopeConflict
- [ ] ShadowedName
- [ ] TransitiveExportError
- [ ] TypesDoNotUnify
- [ ] UndefinedTypeVariable
- [ ] UnknownImport
- [ ] UnknownName
- [ ] WildcardInferredType

### CODE REVIEW

File: errors/NoInstanceFound.md [hunk 1]
```diff
@@ -36,3 +36,66 @@ logGet param = do
 will throw this error. This is because the type signature is not properly constructed, and should account for the `CONSOLE` effect arising from the `log` function. The type checker will not infer that `CONSOLE` should be a part of this type signature in its warning if the type signature is left out, *probably* due to the lifting of the effect. The correct type signature is:
 
 ``` logGet :: forall aff. String -> Aff (ajax :: AJAX, console :: CONSOLE | aff) Unit```
+
+
+### exhaustivity
```
Comment (chexxor) [inline errors/NoInstanceFound.md]: I wonder why exhaustivity was moved here. "NonExhaustivePattern" is no longer an error code?

File: errors/UnknownModule.md [hunk 2]
```diff
@@ -1,36 +0,0 @@
-This error occurs when the type checker knows a token refers to a module, but is unable to find a definition for that module. For example,
```
Comment (chexxor) [inline errors/UnknownModule.md]: These errors are also no longer given by the compiler?

File: errors/ErrorParsingFFIModule.md [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+## Cause
 
-Values should be provided in the form `exports.foo = ...`, similar to a CommonJS module.
+This error occurs when the compiler cannot verify foreign declarations in JavaScript code,
+because it was unable to parse a JavaScript file.
 
-An example can be found [here](https://github.com/purescript/purescript-eff/blob/v0.1.0-rc.1/src/Control/Monad/Eff.js).
+## Fix
+
+- Verify the JavaScript code in your foreign modules.
```
Comment (hdgarrood) [inline errors/ErrorParsingFFIModule.md]: Might be worth mentioning FFI modules must be ES5 here?

File: errors/NoInstanceFound.md (hunk 1 above)
Comment (hdgarrood) [inline errors/NoInstanceFound.md]: Correct: the compiler now generates Partial constraints on inexhaustive pattern matches, and so if a function is declared as not being partial but uses a nonexhaustive pattern match, this becomes a NoInstanceFound error as the compiler will try and fail to find a Partial instance.

File: errors/ErrorParsingFFIModule.md (hunk 3 above)
Comment (paf31) [inline errors/ErrorParsingFFIModule.md]: 👍

**Comment by hdgarrood** [comment]
Yeah, I'd happily lend a hand here if there's an easy way to divide this up. Is the checklist you've done there a list of pages that already exist in this repo, or a complete list of all the errors that the compiler can produce, by the way?

**Comment by paf31** [comment]
@hdgarrood I just added pages for the missing error codes under `missing/`. I'll try to get this merged relatively quickly, and then we can start accepting PRs to fix the remaining pages.

**Comment by paf31** [comment]
@hdgarrood I've moved the pending pages under a new `pending/` directory, so that we don't lose the existing content while we decide what should stay and what should go.

This should hopefully be good to merge now.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**